api/server/router/container/backend.go

```diff
@@ -43,7 +43,7 @@ type stateBackend interface {
 	ContainerStop(name string, seconds int) error
 	ContainerUnpause(name string) error
 	ContainerUpdate(name string, hostConfig *container.HostConfig) ([]string, error)
-	ContainerWait(name string, timeout time.Duration) (int, error)
+	ContainerWait(name string, timeout time.Duration, waitForRemoval bool) (int, error)
```

Comment:
@mlaventure detach event PR is already open; see https://github.com/docker/docker/issues/22898